docs/book/getting-started/deploying-zenml/docker.md

```diff
@@ -1,88 +1,359 @@
+
+```shell
+docker run -it -d -p 8080:80 --name zenml \
+    --add-host host.docker.internal:host-gateway \
+    --env ZENML_STORE_URL=mysql://root:password@host.docker.internal/zenml \
+    zenmldocker/zenml-server
 ```
 
-This command deploys a ZenML server locally in a Docker container. The output shows the URL at which you can access your ZenML dashboard and server.
+Connecting your client to the ZenML server is the same as before:
+
+```shell
+zenml connect --url http://localhost:8080 --username default --password ''
+```
+
+### ZenML server with Docker Compose
+
+Docker compose offers a simpler way of managing multi-container setups on
+your local machine, which is the case for instance if you are looking to deploy
+the ZenML server container and connect it to a MySQL database service also
+running in a Docker container.
+
+To use Docker Compose, you need to [install the docker-compose plugin](https://docs.docker.com/compose/install/linux/)
+on your machine first.
+
+A `docker-compose.yml` file like the one below can be used to start and manage
+the ZenML server container and the MySQL database service all at once:
+
+```yaml
+version: "3.9"
+
+services:
+  mysql:
+    image: mysql:5.7
+    ports:
+      - 3306:3306
+    environment:
+      - MYSQL_ROOT_PASSWORD=password
+  zenml:
+    image: zenmldocker/zenml-server
+    ports:
+      - "8080:80"
+    environment:
+      - ZENML_STORE_URL=mysql://root:password@host.docker.internal/zenml
+      - ZENML_DEFAULT_USERNAME=admin
+      - ZENML_DEFAULT_PASSWORD=zenml
+    links:
+      - mysql
+    depends_on:
+      - mysql
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+    restart: on-failure
+```
+
+Note the following:
+
+- `ZENML_STORE_URL` is set to the special Docker `host.docker.internal` hostname
+to instruct the server to connect to the database over the Docker network. The
+ZenML client knows how to handle the `host.docker.internal` hostname
+differently depending on whether it is running on the host machine or in another
+Docker container.
+- The `extra_hosts` section is needed on Linux to make the `host.docker.internal`
+hostname resolvable from the ZenML server container.
+- This example also uses the `ZENML_DEFAULT_USERNAME` and `ZENML_DEFAULT_PASSWORD`
+environment variables to customize the default account credentials.
+
+To start the containers, run the following command from the directory where
+the `docker-compose.yml` file is located:
+
+```shell
+docker-compose up -d
+```
+
+Connecting your client to the ZenML server is the same as before:
+
+```shell
+zenml connect --url http://localhost:8080 --username default --password ''
+```
+
+## Troubleshooting
+
+You can check the logs of the container to verify if the server is up and depending on where you have deployed it, you can also access the dashboard at a `localhost` port (if running locally) or through some other service that exposes your container to the internet. 
```

Comment:
Wouldn't hurt to explicitly write the logs CLI command here again

Reply:
Done